### PR TITLE
Add new module `odgi squeeze`

### PR DESCRIPTION
**Pull request by subwaystation**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

This adds a new module `odgi squeeze` which is required for https://github.com/nf-core/pangenome.

### CODE REVIEW

File: modules/nf-core/odgi/squeeze/main.nf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    ls *.og >> files
```
Comment (pinin4fjords) [inline modules/nf-core/odgi/squeeze/main.nf:24]:
```suggestion
    ls *.og > files
```
First write should sure to start fresh (not that anything should be getting into the workdir, but still)

File: modules/nf-core/odgi/squeeze/main.nf [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    conda "bioconda::odgi=0.8.2"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/odgi:0.8.2--py39h2add14b_0':
+        'quay.io/biocontainers/odgi:0.8.2--py39h2add14b_0' }"
+
+    input:
+    tuple val(meta), path(graphs)
+
+    output:
+    tuple val(meta), path("*.og"), emit: graph
```
Comment (pinin4fjords) [inline modules/nf-core/odgi/squeeze/main.nf:14]: Can we align the emits?

File: tests/modules/nf-core/odgi/squeeze/main.nf [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+include { ODGI_SQUEEZE } from '../../../../../modules/nf-core/odgi/squeeze/main.nf'
+
+workflow test_odgi_squeeze {
+
+    input = [ [ id:'test', single_end:false ], // meta map
+              [ file(params.test_data['homo_sapiens']['pangenome']['odgi']['pangenome_og'], checkIfExists: true),
+                file(params.test_data['homo_sapiens']['pangenome']['pangenome_smoothxg_gfa'], checkIfExists: true)]
+            ]
+
+    ODGI_SQUEEZE ( input )
```
Comment (pinin4fjords) [inline tests/modules/nf-core/odgi/squeeze/main.nf:14]:
```suggestion
    og_file = file(params.test_data['homo_sapiens']['pangenome']['odgi']['pangenome_og'], checkIfExists: true)
    gfa_file = file(params.test_data['homo_sapiens']['pangenome']['pangenome_smoothxg_gfa'], checkIfExists: true)
    
    ch_input = Channel.from([ 
        [ id:'test', single_end:false ], // meta map
        [ og_file, gfa_file ]
    ])

    ODGI_SQUEEZE ( ch_input )
```

Just a suggestion (obviously tweak as desired)- separating the file definitions makes the logic easier to read.

**Review by pinin4fjords — CHANGES_REQUESTED** [review]
Nice work, very tidy, just a couple of tiny suggestions to improve clarity.

File: modules/nf-core/odgi/squeeze/main.nf (hunk 2 above)
Comment (subwaystation) [reply modules/nf-core/odgi/squeeze/main.nf:14]: Sorry, what is it you want aligned? It is only one emit.

Comment (subwaystation) [reply modules/nf-core/odgi/squeeze/main.nf:14]: Ah, it was not visible in the code shown here. Done.

**Review by pinin4fjords — APPROVED** [review]

**Comment by pinin4fjords** [comment]
> Hmm.

If the fix is non-obvious just revert and merge - all looks good

**Comment by subwaystation** [comment]
Apparently the `Channel.from()` was the problem. Why would that not work, though?

**Comment by subwaystation** [comment]
Because we wanted a Channel of Channels?

**Comment by pinin4fjords** [comment]
> Because we wanted a Channel of Channels?

I should probably have used `of`- sorry! But looking good now.

**Comment by subwaystation** [comment]
Thanks @pinin4fjords !

**Pull request merged**